pymc/tests/test_model_graph.py

```diff
@@ -236,6 +256,10 @@ class TestUnnamedObservedNodes(BaseModelGraphTest):
     model_func = model_unnamed_observed_node
 
 
+class TestObservationDtypeCasting(BaseModelGraphTest):
+    model_func = model_observation_dtype_casting
```

Comment:
Why not just add the function here into the class directly?

Reply:
I just naively followed the structure of previous tests based on `BaseModelGraphTest`...

Reply:
Thanks, I see that now that I expanded past diffs. Dont worry about it on this PR, ill just openly say I dont believe this is the right pattern, it very much feels like we should be using fixtures to reference model setup blocks as pytest provides many more functionality and guarantees.

Thanks for the PR!